### DISCUSSION TITLE
Response to Akka Licensing, and the future of the Play Framework

### DISCUSSION THREAD

**Discussion by BillyAutrey** [playframework/playframework#11405] [Announcements]
The Play Steering committee is fully committed to the open source community, and in maintaining the free and open use of the Play Framework.  As long as this committee exists in its current state, the Play Framework has no intentions of requiring Lightbend licenses or fees to use its software.

The sudden shift in Lightbend's policy on Akka raises many questions about the future of Play.  Many questions have come to the foundation about the future of Play, as well as its usage of Akka.

Our stance is as follows:
1. Akka 2.6 will continue to be open source, and receive CVEs for the next year.  Current Play Framework users should feel safe in using Play and Akka 2.6 interfaces for this duration.
2. Until we can assess the full licensure implications of upgrading to Akka 2.7, the Play Framework will remain on Akka 2.6.
3. To reduce the impact of Play Framework users in the future who might want to use the framework, future versions of Play `2.x` will begin to introduce equivalent APIs for any interfaces that currently only use Akka.  For instance, if the current WebSocket interface will only accept _Akka Streams_ or _Actors_ as an input parameter today, an additional API in `2.x` will expose a non-Akka equivalent behavior.
4. For 3.0, we will consider the removal (or reduction of) Akka interfaces and APIs, to reduce the risk of accidentally incurring a license requirement.  We are also open to removing underlying Akka implementations, if the community expresses interest in collaborating on such a move.
5. We will remove the Lightbend CLA contribution requirement from future Play Framework PRs, and evaluate more open options for its current license structure.

Until we are able to take actions that match this stance, we will continue to work with Lightbend to determine the exact boundaries that result in an Akka license requirement.  And, we will not remove any existing Akka APIs for the immediate future, to ensure that current Akka + Play users are not disrupted.  We hope to be able to share more information in the very near future, once we have our questions answered by Lightbend.

**Comment by SethTisue** [comment]
> Akka 2.6 will [...] receive CVEs for the next year

for those who weren't aware of this change, reference: https://discuss.lightbend.com/t/2-6-x-maintenance-proposal/9949/2

**Comment by mahadi-hossain** [comment]
Removing underlying Akka implementations will be the safest decision. If anyone want to use akka they will use that at their own responsibility.

**Reply by jducoeur** [reply]
In theory, sure -- but there's a lot of Akka inside of Play.  (Much of the past ten years was spent making it more Akka-centric.)  Removing it is a large project, which would require a lot of help.

**Comment by hertg** [comment]
FYI: There is an ongoing effort to get an Akka fork rolling as an Apache incubator project. So replacing Akka with that fork may become a viable option. Probably needs some months to get accepted at the Apache foundation and fully rolling, but there is a lot of support already and [a large amount of willing contributors](https://github.com/mdedetrich/akka-apache-project/discussions/3), including some who worked on Akka previously.

https://github.com/mdedetrich/akka-apache-project/discussions

**Reply by an-tex** [reply]
A switch to Pekko (that's what the fork will be called) is license wise the best option. The tricky part is that Play is not only using Akka under the hood but also exposing it to the end user (at least with [websockets](https://www.playframework.com/documentation/2.8.x/ScalaWebSockets)). This means the Play users would have to migrate their code base to Pekko as well or have two incompatible ActorSystems :(

**Comment by tmccombs** [comment]
Does the play exemption also apply to forks of Play?

**Reply by BillyAutrey** [reply]
You would have to ask Lightbend.  We are not able to make any legal statements regarding licenses granted by another party.
